src/query-builder/SelectQueryBuilder.ts

```diff
@@ -1044,9 +1044,15 @@ export class SelectQueryBuilder<Entity> extends QueryBuilder<Entity> implements
                 transactionStartedByUs = true;
             }
 
-            this.expressionMap.queryEntity = true;
+            if (!this.expressionMap.selectDistinct)
+                this.expressionMap.queryEntity = true;
+
             const results = await this.executeEntitiesAndRawResults(queryRunner);
 
+            // if entities is empty seed with raw
+            if (results.entities.length === 0)
+                results.entities = results.raw;
```

Comment:
also, why do we need this?

Reply:
Ugh yes I should have raised this as a talking point / temporary amendment. This needs a better solution. The results object entities is empty without the queryEntity flag.

Reply:
Can this only be done only when you've set distinct?